src/components/tooltip/Tooltip.tsx

```diff
@@ -138,16 +142,16 @@ class Tooltip extends React.Component<TooltipProps, State> {
         if (isControlled) {
             if (!prevProps.isShown && this.props.isShown) {
                 this.setState({ wasClosedByUser: false });
-            }
-        } else {
-            if (!prevState.isShown && this.state.isShown) {
-                // capture event so that tooltip closes before any other floating components that can be closed by
-                // "Escape" key(e.g. Modal, Menu, etc.)
                 document.addEventListener('keydown', this.handleKeyDown, true);
             }
-            if (prevState.isShown && !this.state.isShown) {
-                document.removeEventListener('keydown', this.handleKeyDown, true);
-            }
+        } else if (!prevState.isShown && this.state.isShown) {
+            // capture event so that tooltip closes before any other floating components that can be closed by
+            // "Escape" key(e.g. Modal, Menu, etc.)
+            document.addEventListener('keydown', this.handleKeyDown, true);
+        }
+
+        if ((prevState.isShown && !this.state.isShown) || (!prevState.wasClosedByUser && this.state.wasClosedByUser)) {
```

Comment:
I might be misunderstanding but I think `prevState.isShown && !this.state.isShown` might also have a similar edge case where this `if` can be entered for a controlled tooltip because the `handleMouseEnter` and `handleMouseLeave` methods: https://github.com/box/box-ui-elements/blob/a603d18f75872864b3b5e822c0d9228f79c5c3fd/src/components/tooltip/Tooltip.tsx#L199-L200

i.e. the tooltip is visible through the `isShown` prop but hovering the mouse could update the state to remove the event listener. not 100% sure though

Reply:
You are right! Thank you for clarifying, I wasn't understanding completely before.